# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/diet-gcc-hack.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

GCC (but potentionally other packages) use -xassembler, which we must not inject
the library objects into. The assembler does not recognize object files ... ,-)

  - Rene Rebe <rene@exactcode.de>

--- dietlibc-0.28/diet.c	2004-12-17 19:47:58.000000000 +0100
+++ dietlibc-0.28-fixed/diet.c	2005-10-11 03:22:19.000000000 +0200
@@ -204,6 +204,8 @@
 	  profile=1;
 	if (!strcmp(argv[i],"-c") || !strcmp(argv[i],"-S"))
 	  compile=1;
+	if (!strcmp(argv[i], "-xassembler"))
+	  goto donttouch;
 	if (!strcmp(argv[i],"-E"))
 pp:
 	  preprocess=compile=1;
